modules/installation-special-config-encrypt-disk-tang.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * installing/install_config/installing-customizing-tang.adoc
+
+[id="installation-encrypting-disks-tang_{context}"]
+== Enabling Tang disk encryption
+Use this procedure to enable Tang mode disk encryption during {product-title} deployment.
+
+.Procedure
+
+. Access a Red Hat Enterprise Linux server from which you can configure the encryption
+settings and run `openshift-install` to install a cluster and `oc` to work with it.
+. Set up or access an existing Tang server. See link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption_security-hardening#network-bound-disk-encryption_configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption[Network-bound disk encryption]
+for instructions. See link:https://youtu.be/2uLKvB8Z5D0[Securing Automated Decryption New Cryptography and Techniques]
+for a presentation on Tang.
+
+. Add kernel arguments to configure networking when you do the RHCOS installations for your cluster.
+If you miss this step, the second boot will fail.
+For example, to configure DHCP networking, identify `ip=dhcp`
+or set static networking when you add parameters to the kernel command line.
+
+. Generate the thumbprint. Install the clevis package, it is not already
+installed, and generate a thumbprint
+from the Tang server. Replace the value of `url` with the Tang server URL:
++
+----
+$ sudo yum install clevis -y
+$ echo nifty random wordwords \
+     | clevis-encrypt-tang \
+       '{"url":"https://tang.example.org"}'
+
+The advertisement contains the following signing keys:
+
+PLjNyRdGw03zlRoGjQYMahSZGu9
+
+Do you wish to trust these keys? [ynYN] y
+eyJhbmc3SlRyMXpPenc3ajhEQ01tZVJiTi1oM...
+----
+. Create a B64 encoded file, replacing the URL of the Tang server (`url`) and thumbprint (`thp`) you just generated:
++
+----
+$ (cat <<EOM
+{
+ "url": "https://tang.example.com",
+ "thp": "PLjNyRdGw03zlRoGjQYMahSZGu9"
+}
+EOM
+) | base64 -w0
+
+ewogInVybCI6ICJodHRwczovL3RhbmcuZXhhbXBsZS5jb20iLAogInRocCI6ICJaUk1leTFjR3cwN3psVExHYlhuUWFoUzBHdTAiCn0K
+----
+
+. Replace the “source” in the TPM2 example with the B64 encoded file using the following format:
```

Comment:
If the snippet in installation-special-config-encrypt-disk.adoc for enabling TPM via MachineConfig is changed, this should also be updated to a MachineConfig.

Reply:
Done